### ISSUE TITLE
Non-blocking stdin

### ISSUE THREAD

**Issue by denik** [gevent/gevent#6]
<b>What steps will reproduce the problem?</b>
1. use sys.stdin.readline()

<b>What is the expected output? What do you see instead?</b>
Expected that other greenlet continue to run while readline() blocks.
Instead, the whole process is blocked.

<p><a href="http://code.google.com/p/gevent/issues/detail?id=6">Reported by Denis.Bilenko.</a></p><hr/><h2>earlier comments</h2>
<p><strong>ferenczi.viktor said, at 2011-02-12T15:28:54.000Z:</strong></p>

<p>There is an example here, not exactly for your use case, but very close:

https://bitbucket.org/denis/gevent/src/23fd80c3be00/examples/processes.py

It might be written by you, actually.

Could we use the same fcntl call to make the process' own stdin non-blocking? It would allow for reimplementing readline in a cooperative way.

It won't work on Windows, however, since no way to wait on unnamed pipes without introducing threads. On Windows the only possibility is to use a reader thread for the blocking readline call, which is ugly, but better than nothing.</p><p><strong>Denis.Bilenko said, at 2012-07-10T09:39:57.000Z:</strong></p>

<p>You can do it now:

``` python
from gevent.fileobject import FileObject

# This makes stdin's FD non-blocking and replaces sys.stdin with
# a wrapper that is integrated into the event loop
sys.stdin = FileObject(sys.stdin)

for line in sys.stdin:
  ...
```

**Comment by ryanchou1991** [comment]
I have same issue on non-blocking operations on FileObject. My program has spawning multi gevent threads. All of them have written on the same file descriptor. While I recognized that it will do harmful to the performance. Does anybody could offer me some suggestions on this issue? 
FileObjectThread?

**Comment by Ivoz** [comment]
@ryanchou1991 I'd recommend you ask in a new thread/issue

**Comment by ryanchou1991** [comment]
@lvoz OK， I will create a new issue :) Thanks
